### PR TITLE
Fea/d06

### PR DESCRIPTION
**Pull request by taherkheli**

### CODE REVIEW

File: src/D06/Group.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+namespace aoc.D06
+{
+	public class Group
+	{
+		List<HashSet<char>> _responses = new List<HashSet<char>>();
+
+		public Group(string[] lines)
+		{
+			foreach (var line in lines)
+				_responses.Add(new HashSet<char>(line.ToCharArray()));
```
Comment (Jaxwood) [inline src/D06/Group.cs:12]: a string is an alias for char[] as far as I remember. No need for ToCharArray.

File: src/Shared/Lib.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+namespace aoc.Shared
+{
+	public static class Lib
+	{
+		public static List<List<string>> GetEmptyLineSeparatedChunks(string [] lines)
+		{
+			var chunks = new List<List<string>>();
+			var chunk = new List<string>();		
+
+			for (int i = 0; i < lines.Length; i++)
```
Comment (Jaxwood) [inline src/Shared/Lib.cs:13]: why not a `foreach`

File: src/D06/Group.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		}
+
+		public static int GetAnyoneAnsweredYesCount(List<Group> groups)
```
Comment (Jaxwood) [inline src/D06/Group.cs:15]: I would use IEnumerable instead of List - to avoid mutation

File: src/Shared/Lib.cs (hunk 2 above)
Comment (taherkheli) [reply src/Shared/Lib.cs:13]: @Jaxwood I need special processing at last line. How do you check if its last line in a  foreach?

File: src/D06/Group.cs (hunk 3 above)
Comment (taherkheli) [reply src/D06/Group.cs:15]: @Jaxwood I changed the signature of both methods to `IEnumerable` but how doe sit improve anything? please elaborate how it impacts immutability

**Pull request merged**